src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -21,6 +21,8 @@ public class SarifVersionOneToCurrentVisitor : SarifRewritingVisitorVersionOne
         private Run _currentRun;
         private RunVersionOne _currentV1Run;
         private int _threadFlowLocationNestingLevel;
+        private IDictionary<string, int> _v1FileKeytoV2IndexMap;
+        private IDictionary<Tuple<Uri, string>, int> _v2FileToIndexMap;
```

Comment:
>Tuple [](start = 28, length = 5)

I introduce this `Tuple` (which holds a `uri`, `uriBaseId` pair), rather than using `FileLocation` itself as the dictionary key. This is because when I create a new `FileLocation` object and try to decide if it's present in `run.files`, the `fileIndex` properties will never match. I just want to compare `uri` and `uriBaseId`. I could also have created a comparer.

I will add a comment about this.

Reply:
Possibly. Let me think about it.

---
In reply to: [248879799](https://github.com/Microsoft/sarif-sdk/pull/1221#discussion_r248879799) [](ancestors = 248879799,248877564)